### PR TITLE
SyncBot: Synchronise Front conversation states into Flowdock

### PR DESCRIPTION
**Pull request by sqweelygig**
- Retype message information into event information to support a broader scope of events
- Add a dictionary of state flags to set/clear to the event information
- Add handling for thread events that specify a change of state
- Pass change of state on Front through to SyncBot
- Add instructions for new Front events to emit
- Store a thread's state in Flowdock as initial message tags
- Explicitly require message and details to encode when stringifying metadata
- Make log line fall back onto thread title
- Put intercomHack into the property pertaining to the current service, rather than the payload
- While I was there my IDE told the Front Session could be readonly and that a regex had redundant escape characters

Change-type: patch
Signed-off-by: Andrew Lucas <andrewl@resin.io>
Connects-to: #564

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [ ] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #564

### CODE REVIEW

**Comment by sqweelygig** [comment]
@lekkas is this a technical as well as UX approve?

**Comment by lekkas** [comment]
This is only a UX approval, as far as I can see there are no UX changes other than the tags, is that correct?

**Comment by sqweelygig** [comment]
Ack to nature of review, and correct to nature of changes.

File: lib/services/messenger-types.d.ts [hunk 1]
```diff
@@ -42,32 +43,47 @@ export interface ReceiptIds extends MessengerBaseIds {
 	message: string;
 	/** The ID of the thread. */
 	thread: string;
+	// https://github.com/resin-io-modules/resin-procbots/issues/301
+	/**
+	 * For Intercom messages on Front, whether they had an empty subject line
+	 * An empty subject line indicates the message was inputted via the Front or Intercom UI
+	 * A subject line indicates the message was inputted via the Front API
+	 * Needed because Front mangles the message into plaintext and refuses to let you see raw
```
Comment (hedss) [inline lib/services/messenger-types.d.ts]: Sigh. :(

Comment (sqweelygig) [reply lib/services/messenger-types.d.ts]: It'll be better in Jellyfish.  At the moment I'm integrating quirky with quirky via quirky, which is a big pain.  Under Jellyfish I'll at least be integrating quirky with quirky via **something we control**.

**Pull request merged**